tt_metal/impl/program/program.hpp

```diff
@@ -77,11 +77,13 @@ class Program {
     Program(Program &&other) = default;
     Program& operator=(Program &&other) = default;
 
+    void set_global_id(uint64_t id);
```

Comment:
Is this only called by ttnn and tteager? So other ops and kernels don't have a global id?

Reply:
Right, this is for use by higher levels. Making direct tt_metal calls can ignore this and op field will show up as zero ...

Reply:
i.e. this is for assigning unique ids for op_report

Reply:
Where is it initialized to 0?

Reply:
Good find, I thought I added it. Will add to initializer list for program

Reply:
Added